app/controllers/submissions_controller.rb

```diff
@@ -22,12 +22,12 @@ def create
     @ontology.update
     
     @submission_saved = @submission.save
-    if !@submission_saved || @submission_saved.errors
+    if response_error?(@submission_saved)
       @errors = response_errors(@submission_saved) # see application_controller::response_errors
 
-      if @errors[:error][:uploadFilePath]
+      if @errors[:error] && @errors[:error][:uploadFilePath]
```

Comment:
This code doesn't appear to work as intended. I tested both cases where the uploadFilePath and contact parameters are empty. There's no `:error` key in `@errors`, so the if conditions fail and the more human readable error messages aren't used.

Reply:
I've committed a fix for this locally